### DISCUSSION TITLE
Tag management

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8104] [Ideas] [answered]
> Created by: Brandon Kelly (brandon@pixelandtonic.com) on 2015/03/06 03:54:45 +0000
> Votes at time of UserVoice import: 81

-----------------------------------

It would be nice if there was a place in the CP where we could manage all of the existing tags.

**Comment by angrybrad** [comment]
> Posted by Alex Woodman (neverfable@gmail.com) on 2016/10/26 20:02:32 +0000
-----------------------------------

This needs to be part of core! Doesn&#39;t seem difficult to implement either, and it seems like an awfully common issue.

**Comment by angrybrad** [comment]
> Posted by Elliot Lewis (elliot@notwothesame.com) on 2016/03/17 09:23:34 +0000
-----------------------------------

Bob Olde Hampsink plugin is perfect. Buy him a few cocktail and roll it in!

**Comment by angrybrad** [comment]
> Posted by Ben Parizek (ben@barrelstrengthdesign.com) on 2015/03/12 23:19:32 +0000
-----------------------------------

Alternatively, maybe Tags should just become a plugin and not part of Core.  It feels like it hasn&#39;t been given much love right now.

**Comment by angrybrad** [comment]
> Posted by Rob (rob@layermade.com) on 2015/07/19 22:22:24 +0000
-----------------------------------

I&#39;d go further and say we need more control over tags (similar to user relations and blocks &quot;limit&quot; box) : http://bit.ly/1VqvlIJ

**Comment by angrybrad** [comment]
> Posted by Bob Olde Hampsink (b.oldehampsink@itmundi.nl) on 2015/04/30 06:39:01 +0000
-----------------------------------

As long as this isn&#39;t part of Craft&#39;s core (yet), there is a Tag Manager plugin here https://github.com/boboldehampsink/tagmanager

**Comment by whoisgina** [comment]
Would love to see this become part of core! For now we're going to try to get Tag Manager to work with Craft 3. Has anyone had any luck with that?

**Comment by brandonkelly** [comment]
@whoisgina I don’t think @boboldehampsink has ported it to Craft 3 yet (not listed on https://github.com/craftcms/plugins). Maybe he’s hoping we’ll add it to Craft ourselves 🙃

**Comment by mattandrews** [comment]
I'd love to see this happen – it's the only missing feature in Craft for my usecase. The plugin above is deprecated now so currently the only way I can edit a tag (which we're using as "authors" for a blogpost entry type) is to find a blog written by that author, and double click their name in the tag list.

**Comment by khalwat** [comment]
I'm hoping you'll add it to Craft yourselves, FWIW 🙃

**Comment by brandonkelly** [comment]
We’re going to add it in a 3.x release, but not sure exactly when yet.

**Comment by ipetrov87** [comment]
I hope it will be soon.

**Comment by brandonkelly** [comment]
Ether Creative just released a new tag management plugin: https://github.com/ethercreative/tags

Should work well while you wait for a built-in solution.

**Comment by kmgdevelopment** [comment]
Ether's plugin hasn't been updated in 6 months and issues go unanswered.

Tag management in Core would very much be appreciated.

**Comment by ryanpcmcquen** [comment]
Is this being worked on?

**Comment by brandonkelly** [comment]
No, the plan is to do away with Tags and Categories entirely, merging those concepts with Entries (and giving Entires field all the same features that Tags and Categories fields have currently). Plus make the Entries index page more customizable so you can still present certain sections as “categories” or “tags” if you want.

**Comment by kmgdevelopment** [comment]
I'm fine with that. Been using Categories to manage tags for a while and it's been working good, and honestly like Brandon said Categories are basically just channel-type sections anyway with slight UI differences.

**Comment by vnali** [comment]
> No, the plan is to do away with Tags and Categories entirely, merging those concepts with Entries (and giving Entires field all the same features that Tags and Categories fields have currently). Plus make the Entries index page more customizable so you can still present certain sections as “categories” or “tags” if you want.

@brandonkelly 
so in future isn't it possible to add a tag field to custom elements via field layout for custom plugins :( ?  
IMHO maybe checkbox or radio field provide kind of same functionality, but tag provide far better UI.

**Comment by brandonkelly** [comment]
Giving Entries fields the ability have similar UIs as Categories and Tags fields is the main thing I meant by that.

**Comment by vnali** [comment]
@brandonkelly , so if  i use tag field in a **custom element type** to label  those custom elements -and this custom element type is not about entries at all-, can i do the same thing and tag my custom elements with entries field in future?

**Comment by brandonkelly** [comment]
@vnali Yes, there won’t be any authoring feature regressions with this change.

**Comment by Friksel** [comment]
@brandonkelly any news on this? Are these plans for v3.x or v4.x? And will there be a way to create new tags directly from within the field like we can now?

**Comment by brandonkelly** [comment]
@Friksel We’ve started planning it out a bit, but no news yet. Yes, part of the plan will be to add new UI modes to Entries fields, so they can behave like Tags/Categories fields do currently.

**Comment by jamiematrix** [comment]
Did anything come from this? We have a client who has 2 tags they want to remove - currently I think it's a bit much to install a plugin to remove the 2 tags

**Reply by jamiematrix** [reply]
I've seen it's not recomended to delete tags from the DB directly (from the elements table) as it's added in multiple places, but the [Tags ](https://github.com/ethercreative/tags) plugin has an `actionDelete` function that appears to do just that ... `!Craft::$app->elements->deleteElement($tag)` unless I'm reading that wrong?
If it's safe to do this would it be OK to create a temporary PHP file and manually add the tag ID in to be deleted?

**Reply by brandonkelly** [reply]
I’ve just [added](https://github.com/craftcms/cms/commit/795e957620bddbd526a9bce52aff136d98295516) a new `elements/delete` command for Craft 4.1 (coming tomorrow), which you will be able to use to quickly delete any element:

```sh
php craft elements/delete 1234
```

**Comment by kellyannmcnamara01** [comment]
I see that Brandon, you have added a command for Craft, but our clients aren't technically savvy and should be able to manage the tags of their own CMS. That's our main selling point to our clients for Craft CMS, we build it, and they control it.

Currently, we are only able to add tags if it's been attached to a field, which is rather funny (but annoying to our clients) – and we can't edit or delete any within the admin.

It would be most ideal if once a tags group was created, the admin sidebar would have a "Tags" option, once clicked you could select the tag group and see all the tags entered within it. It would be ideal to be CRUD said tags in this view, like [the old deprecated plugin](https://github.com/boboldehampsink/tagmanager) used to.

**Reply by brandonkelly** [reply]
Yeah that command wasn’t meant to resolve this discussion at all :)

**Reply by kellyannmcnamara01** [reply]
Okay, cool – that's what I figured. Will a feature like I mentioned above be released?

**Reply by brandonkelly** [reply]
Most likely we are just going to end up merging the concepts of entries, categories, and tags together, and give Entries fields the option to choose which of those three types of UI modes they want.

**Reply by kellyannmcnamara01** [reply]
Thank you for your explanations

**Reply by farmbureau** [reply]
I'm in the same situation, client loves tags, needs to manage tags, thank you for your time and attention

**Reply by kellyannmcnamara01** [reply]
@brandonkelly, I just want to follow up on this b/c it's been a couple of months. We are about to release a client site that I have been holding back due to the client's inability to see and edit their tags in the tag groups in the admin. I was wondering if you have pushed an alternative or have a suggestion otherwise. If not I am going to need to rethink a massive amount of functional work done for the site.

Please note functionality, we are using tags to auto-populate related suggested entries in the front end, as well as tags in our various in-depth search queries.

**Reply by brandonkelly** [reply]
@kellyannmcnamara01 See https://github.com/craftcms/cms/discussions/8104#discussioncomment-3964947.

> If not I am going to need to rethink a massive amount of functional work done for the site.

In theory it shouldn’t be too much work to update your PHP code / Twig templates to refer to the new sections representing your old tag groups, as the content models should basically be identical. Do you have any code that is especially concerning?

**Comment by murtaugh** [comment]
Just a bump on this — I'm working on my first Craft project and it's surprising to me that Tags have been given such short shrift in the control panel. Tags are no longer as important as they once were, but they're still very useful, especially for organizing editorial collections of content.

**Reply by proimage** [reply]
I'd say that Tags may no longer be as _novel_ as they used to be, but they're still quite important. Even this thread here on GH is tagged ("labeled"). ¯\_(ツ)_/¯

**Comment by brandonkelly** [comment, marked as answer]
We have decided to officially retire the concept of tags (and categories and global sets) in Craft 5. Here’s an extended explanation of how that will play out: https://github.com/craftcms/cms/discussions/9781#discussioncomment-3964944
